crates/weaver_forge/src/extensions/otel.rs

```diff
@@ -828,6 +829,162 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_deprecated() {
+        #[derive(Serialize)]
+        struct Ctx {
+            attr: AttributeSpec,
+        }
+
+        let mut env = Environment::new();
+        let attr_yaml = r#"
+            id: attr
+            type: string
+            brief: This attribute is replaced by new_name.
+            note: A note.
+            requirement_level: required
+            deprecated: Replaced by new_name.
+        "#;
+        let attr: AttributeSpec = serde_yaml::from_str(attr_yaml).unwrap();
+
+        otel::add_filters(&mut env);
+        otel::add_tests(&mut env);
+
+        assert_eq!(
+            env.render_str(
+                "{{ attr.deprecated.reason }}: {{ attr.deprecated.note }}",
+                Ctx { attr },
+            )
+            .unwrap(),
+            "uncategorized: Replaced by new_name."
+        );
+
+        // ---------------------------------------------------------------------
+        let attr_yaml = r#"
+            id: attr
+            type: string
+            brief: This attribute is replaced by new_name.
+            note: A note.
+            requirement_level: required
+            deprecated:
+                reason: renamed
+                renamed_to: new_name
+        "#;
+        let attr: AttributeSpec = serde_yaml::from_str(attr_yaml).unwrap();
+        assert_eq!(
+            env.render_str(
+                "{{ attr.deprecated.reason }}: {{ attr.deprecated.note }}",
+                Ctx { attr },
+            )
+            .unwrap(),
+            "renamed: Replaced by `new_name`."
```

Comment:
I don't know if `renamed` helps anyone. Can we do 

```suggestion
            "Replaced by `new_name`."
```

? It'd match what we did in semconv so far

Reply:
In this unit test, I'm rendering this Jinja expression `{{ attr.deprecated.reason }}: {{ attr.deprecated.note }}`, so the `renamed: ` is something that is not part of the `deprecated.note` but used to check 2 fields in a unit test.